src/test/java/org/dbos/apiary/xa/XATests.java

```diff
@@ -1,45 +1,164 @@
 package org.dbos.apiary.xa;
 
 import com.google.protobuf.InvalidProtocolBufferException;
+
+import io.netty.util.internal.ThreadLocalRandom;
+
 import org.dbos.apiary.client.ApiaryWorkerClient;
 import org.dbos.apiary.utilities.ApiaryConfig;
 import org.dbos.apiary.worker.ApiaryNaiveScheduler;
 import org.dbos.apiary.worker.ApiaryWorker;
+import org.dbos.apiary.xa.procedures.BankAudit;
+import org.dbos.apiary.xa.procedures.BankTransfer;
 import org.dbos.apiary.xa.procedures.GetApiaryClientID;
+import org.dbos.apiary.xa.procedures.MySQLXAQueryPerson;
+import org.dbos.apiary.xa.procedures.PostgresXAQueryPerson;
+import org.dbos.apiary.xa.procedures.XAQueryPersonBoth;
 import org.dbos.apiary.xa.procedures.XASimpleTest;
+import org.dbos.apiary.xa.procedures.XAUpsertPerson;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.postgresql.xa.PGXAConnection;
+import org.postgresql.xa.PGXAException;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 public class XATests {
     private static final Logger logger = LoggerFactory.getLogger(XATests.class);
 
     private ApiaryWorker apiaryWorker;
 
-    @BeforeEach
-    public void resetTables() {
-
-    }
-
     @AfterEach
     public void cleanupWorker() {
         if (apiaryWorker != null) {
             apiaryWorker.shutdown();
         }
     }
 
+    @Test
+    public void testSimpleMySQL() throws InvalidProtocolBufferException {
+        logger.info("testSimpleMySQL");
+        try {
+            MySQLXAConnection conn = new MySQLXAConnection("172.17.0.1", 3306, "dbos", "root", "dbos");
+            conn.dropTable("test");
```

Comment:
I think the IP address here could just be localhost(?) Anyways, may consider making the IP address a variable not hardcoded.

Reply:
``172.17.0.1`` is a private IP address only for the docker container. I think it is fine.

Reply:
The tricky thing is the Docker IP could change, if there are other containers running in the system. Does `localhost` cause any error?

Reply:
Fixed. ``localhost`` works.